tmc-plugin/src/fi/helsinki/cs/tmc/actions/DownloadAdaptiveExerciseAction.java

```diff
@@ -55,6 +58,12 @@ public void bgTaskReady(Exercise ex) {
                 dialogs.displayMessage("Adaptive exercise downloaded");
                 TmcProjectInfo proj = projectMediator.tryGetProjectForExercise(ex);
                 projectMediator.openProject(proj);
+
+                if (courseDb.getCurrentCourseExercises().contains(ex)) {
+                    courseDb.getCurrentCourseExercises().remove(ex);
+                }
+                courseDb.getCurrentCourseExercises().add(ex);
+                courseDb.exerciseDownloaded(ex);
```

Comment:
I assume this somehow refreshes the exercise's state or something? Add a comment that details what this operation (removing and re-adding the exercise) achieves. Otherwise it just seems like a NOP.